### PR TITLE
feat: JSON logging for Gunicorn

### LINKED ISSUE #3637: Make `LOG_FORMAT` work for access logs, allow custom access log format
### Is your feature request related to a problem? Please describe.

Currently, access log format is hardcoded.

This prevents users from customising their access log output.

Additionally, setting `LOG_FORMAT` does not apply the JSON formatting to the access logs.

### Describe the solution you'd like.

Implement the following:

1. Support setting custom logconfigs — either [logconfig](https://docs.gunicorn.org/en/stable/settings.html#logconfig), [logconfig_dict](https://docs.gunicorn.org/en/stable/settings.html#logconfig-dict) or [logconfig_json](https://docs.gunicorn.org/en/stable/settings.html#logconfig-json) command line options.
2. Set up a log formatter (e.g. this [JSON formatter](https://til.codeinthehole.com/posts/how-to-get-gunicorn-to-log-as-json/)) depending on how `LOG_FORMAT` is set.

### Describe alternatives you've considered

N/A

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by khvn26**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

This PR does the following:
- Move default Gunicorn settings to a `gunicorn.conf.py` file. This allows users to override Gunicorn settings using the `GUNICORN_CMD_ARGS` environment variable.
- Add `GunicornAccessLogJsonFormatter` and `GunicornJsonLogger` to make Gunicorn respect the `LOG_FORMAT` setting.
- Expose the `ACCESS_LOG_FORMAT` environment variable.
- `make serve` now configures Gunicorn with `gunicorn.conf.py`.

Closes #3637.

## How did you test this code?

Ran `make serve` locally with `LOG_FORMAT`, `ACCESS_LOG_FORMAT` and `ACCESS_LOG_LOCATION` set to various values.

### CODE REVIEW

File: api/util/logging.py [hunk 1]
```diff
@@ -1,27 +1,73 @@
...
+        return {
+            "levelname": record.levelname,
+            "message": formatted_message,
+            "timestamp": self.formatTime(record, self.datefmt),
+            "logger_name": record.name,
+            "process_id": record.process,
+            "thread_name": record.threadName,
+        }
+
+    def format(self, record: logging.LogRecord) -> str:
```
Comment (zachaysan) [inline api/util/logging.py:26]: The docstring of this function is now out of date.

File: api/util/logging.py [hunk 2]
```diff
@@ -1,27 +1,73 @@
...
+            "referer": record.args["f"],
+            "duration_in_ms": record.args["M"],
+            "pid": record.args["p"],
+        }
+
+
+class GunicornJsonLogger(GunicornLogger):
+    def setup(self, cfg: Config) -> None:
+        super().setup(cfg)
+        if settings.LOG_FORMAT == "json":
```
Comment (zachaysan) [inline api/util/logging.py:57]: If it's not set to JSON then the logger won't really be a JSON logger, right? It's weird to let it operate as a normal logger with the name of the class `GunicornJsonLogger`

File: docs/docs/deployment/hosting/locally-api.md [hunk 3]
```diff
@@ -150,7 +150,11 @@ the below variables will be ignored.
   define a shared DJANGO_SECRET_KEY.
 - `LOG_LEVEL`: DJANGO logging level. Can be one of `DEBUG`, `INFO`, `WARNING`, `ERROR`, `CRITICAL`
 - `LOG_FORMAT`: Can be `generic` (plain-text) or `json`. Defaults to `generic`.
-- `ACCESS_LOG_LOCATION`: The location to store web logs generated by gunicorn if running as a Docker container. If not
+- `GUNICORN_CMD_ARGS`: Gunicorn command line arguments. Overrides Flagsmith's defaults. See
+  [Gunicorn documentation](https://docs.gunicorn.org/en/stable/settings.html) for reference.
+- `ACCESS_LOG_FORMAT`: Message format for Gunicorn's access log. See
+  [variable details](https://docs.gunicorn.org/en/stable/settings.html#access-log-format) to define your own format.
+- `ACCESS_LOG_LOCATION`: The location to store web logs generated by Gunicorn if running as a Docker container. If not
   set, no logs will be stored. If set to `-` the logs will be sent to `stdout`.
```
Comment (zachaysan) [inline docs/docs/deployment/hosting/locally-api.md]: > If not set, no logs will be stored.

I think if no logs will be stored then `stdout` will actually be used, which is what we should say here, no?

**Review by zachaysan — APPROVED** [review]
Overall looks good, just some minor comments.

File: docs/docs/deployment/hosting/locally-api.md (hunk 3 above)
Comment (khvn26) [reply docs/docs/deployment/hosting/locally-api.md]: Thanks! Updated the docs.

File: api/util/logging.py (hunk 2 above)
Comment (khvn26) [reply api/util/logging.py:57]: Renamed to `GunicornJsonCapableLogger`.

File: api/util/logging.py (hunk 1 above)
Comment (khvn26) [reply api/util/logging.py:26]: Thanks, removed!

**Pull request merged**